### PR TITLE
API 1.1 ongoing review PR

### PR DESCRIPTION
**Pull request by mattheworiordan**
We will use this PR to track ongoing proposed changes for the 1.1 specification.  Commits merged in are generally accepted to be included in 1.1

### CODE REVIEW

**Comment by SimonWoolf** [comment]
Rebased on latest master (somewhat painfully) and merged in #306, https://github.com/ably/docs/pull/293, https://github.com/ably/docs/pull/186

File: content/client-lib-development-guide/features.textile [hunk 1]
```diff
@@ -189,7 +201,7 @@ h3(#rest-channel). Channel
...
 ** @(RSL1f)@ Unidentified clients using "Basic Auth":https://en.wikipedia.org/wiki/Basic_access_authentication (i.e. any @clientId@ is permitted as no @clientId@ specified):
-** @(RSL1g)@ If the total size of the message or (if publishing an array) messages, calculated per "TO3l7":#TO3l7, exceeds the @maxMessageSize@, then the client library should reject the publish and indicate an error
+** @(RSL1g)@ If the total size of the message or (if publishing an array) messages, calculated per "TO3l8":#TO3l8, exceeds the @maxMessageSize@, then the client library should reject the publish and indicate an error
 *** @(RSL1f1)@ When a @Message@ with a @clientId@ value is published, Ably will accept and publish that message with the provided @clientId@. A test should assert via the history API that the @clientId@ of the published @Message@ is populated
 ** @(RSL1g)@ Identified clients with a @clientId@ (as a result of either an explicitly configured @clientId@ in @ClientOptions@, or implicitly through Token Auth):
```
Comment (funkyboy) [inline content/client-lib-development-guide/features.textile]: I struggle to reposition this, duplicate of https://github.com/ably/docs/pull/258/files#diff-11900b395df266bc2bbfc1d11bdfc7a0R204

**Comment by mattheworiordan** [comment]
@paddybyers if we are to merge this with Push I think we should:

* Make it clear exception reporting is an optional feature in client libs
* We need to remove the various unrelated commits 
* We should add extras to PresenceMessage for consistency (it's one field, so now is a good time for that)
* Close https://github.com/ably/docs/issues/339
* Decide if https://github.com/ably/docs/issues/335 is still needed
* Confirm if https://github.com/ably/docs/issues/328 is done
* Consider https://github.com/ably/docs/issues/280 to ensure 1.1 is more stable moving forwards.  All libs should do this now anyway so we're not adding complexity, just tightening up the requirements
* Close https://github.com/ably/docs/issues/242
* Consider https://github.com/ably/docs/issues/232 given it's a one liner in all libraries but can save lots of headaches for developers.  If we follow a standard URL format for errors now we can save a lot of dev headaches, and it's easy to implement. I don't think we should wait for 1.2, that could be 6 months away or more,.
* I think https://github.com/ably/docs/issues/224 is mostly addressed now? 
* I think https://github.com/ably/docs/issues/75 is resolved with this work?
* Support transient publishing https://github.com/ably/docs/issues/284, even if we mark this as optional.  At least it will be in the spec and finalised.  I think we can ignore attach flags as it should be unrelated and we just need a simple API for publishing over a realtime connection.

**Comment by paddybyers** [comment]
@SimonWoolf @funkyboy any comments on the above?

**Comment by funkyboy** [comment]
> Make it clear exception reporting is an optional feature in client libs

**Comment by SimonWoolf** [comment]
> We need to remove the various unrelated commits

Wait, which commits do you want to remove from this PR? Commits that have already been merged into this pr are ones we've previously agreed should definitely go into 1.1, would be surprising to remove any now

> Consider #280 to ensure 1.1 is more stable moving forwards.

:+1: agree

> Consider #232 given it's a one liner in all libraries but can save lots of headaches for developers

:+1: agree

> I think #224 is mostly addressed now?

No commits re that are in here, but could be done with a tiny pr referencing the rfc

> I think #75 is resolved with this work?

Yup, the fixes for that are already merged into here

> Support transient publishing #284, even if we mark this as optional.

Transient publishes [got merged](https://github.com/ably/docs/pull/323) into the 1.0 spec on the grounds that we wanted to allow client libraries  (in particular ably-js) to implement it immediately. So it's one of several things that is in 1.0 but after the `v1.0` tag so should still be included in the 1.0->1.1 spec diff. 

> Optional as in our customers can flip a switch to turn it on?

I believe Matt meant optional in the sense that client libraries do not have to implement it in order to be considered to conform to the 1.1 spec

**Comment by mattheworiordan** [comment]
>> We need to remove the various unrelated commits

> Wait, which commits do you want to remove from this PR? Commits that have already been merged into this pr are ones we've previously agreed should definitely go into 1.1, would be surprising to remove any now

See commits a49056e...79ffa18

>> Optional as in our customers can flip a switch to turn it on?

> I believe Matt meant optional in the sense that client libraries do not have to implement it in order to be considered to conform to the 1.1 spec

Yes

**Comment by paddybyers** [comment]
> Make it clear exception reporting is an optional feature in client libs

https://github.com/ably/docs/pull/408

> We need to remove the various unrelated commits

Done, in https://github.com/ably/docs/tree/integration-spec-1-1

> We should add extras to PresenceMessage for consistency (it's one field, so now is a good time for that)

https://github.com/ably/docs/pull/409

> Close #339

In progress

> Decide if #335 is still needed

I'm proposing to defer (https://github.com/ably/docs/issues/335#issuecomment-377074900)

> Confirm if #328 is done

Yes

> Consider #280 to ensure 1.1 is more stable moving forwards. All libs should do this now anyway so we're not adding complexity, just tightening up the requirements

https://github.com/ably/docs/pull/411

> Close #242

Done

> Consider #232 given it's a one liner in all libraries but can save lots of headaches for developers. If we follow a standard URL format for errors now we can save a lot of dev headaches, and it's easy to implement. I don't think we should wait for 1.2, that could be 6 months away or more,.

https://github.com/ably/docs/pull/412

> I think #224 is mostly addressed now?

https://github.com/ably/docs/pull/410

> I think #75 is resolved with this work?

Yes

> Support transient publishing #284, even if we mark this as optional. At least it will be in the spec and finalised. I think we can ignore attach flags as it should be unrelated and we just need a simple API for publishing over a realtime connection.

Done already

**Comment by SimonWoolf** [comment]
I've now combined master-1-1 and integration-spec-1-1 and promoted the result to master. The previous master is now https://github.com/ably/docs/tree/v1.0-as-amended

**Pull request closed without merging**